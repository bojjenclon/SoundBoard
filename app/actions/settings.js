// @flow
export const SET_MAX_CACHE_SIZE = 'SET_MAX_CACHE_SIZE';
export const SET_GRID_COLUMNS = 'SET_GRID_COLUMNS';
export const SET_ALLOW_CONCURRENT_AUDIO = 'SET_ALLOW_CONCURRENT_AUDIO';

export function setMaxCacheSize(value: number) {
  return {
    type: SET_MAX_CACHE_SIZE,
    value
  };
}

export function setGridColumns(value: number) {
  return {
    type: SET_GRID_COLUMNS,
    value
  };
}

export function setAllowConcurrentAudio(value: boolean) {
  return {
    type: SET_ALLOW_CONCURRENT_AUDIO,
    value
  };
}
