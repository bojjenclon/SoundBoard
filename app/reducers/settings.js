// @flow
type State = {
  maxCacheSize: number,
  gridColumns: number,
  allowConcurrentAudio: boolean
};

type Action = {
  type: string,
  // eslint-disable-next-line flowtype/no-weak-types
  value: any
};

const defaultState: State = {
  maxCacheSize: 25,
  gridColumns: 6,
  allowConcurrentAudio: false
};

const settings = (state: State = defaultState, action: Action) => {
  switch (action.type) {
    case 'SET_MAX_CACHE_SIZE':
      return {
        ...state,

        maxCacheSize: action.value
      };
    case 'SET_GRID_COLUMNS':
      return {
        ...state,

        gridColumns: action.value
      };
    case 'SET_ALLOW_CONCURRENT_AUDIO':
      return {
        ...state,

        allowConcurrentAudio: action.value
      };
    default:
      return state;
  }
};

export default settings;
