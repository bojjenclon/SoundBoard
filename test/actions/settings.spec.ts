import { spy } from 'sinon';
import * as actions from '../../app/actions/settings';

describe('actions', () => {
  it('should create max cache size setter action with appropriate value', () => {
    expect(actions.setMaxCacheSize(15)).toMatchSnapshot();
  });

  it('should create grid columns setter action with appropriate value', () => {
    expect(actions.setGridColumns(3)).toMatchSnapshot();
  });

  it('should create allow concurrent audio setter action with appropriate value', () => {
    expect(actions.setAllowConcurrentAudio(true)).toMatchSnapshot();
  });
});
