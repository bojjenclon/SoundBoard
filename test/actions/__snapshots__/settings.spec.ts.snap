// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create max cache size setter action with appropriate value 1`] = `
Object {
  "type": "SET_MAX_CACHE_SIZE",
  "value": 15
}
`;

exports[`actions should create grid columns setter action with appropriate value 1`] = `
Object {
  "type": "SET_GRID_COLUMNS",
  "value": 3
}
`;

exports[`actions should create allow concurrent audio setter action with appropriate value 1`] = `
Object {
  "type": "SET_ALLOW_CONCURRENT_AUDIO",
  "value": true
}
`;
